### PR TITLE
'allocation_content_length'

### PR DESCRIPTION
**Pull request by wind-Lv**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
https://travis-ci.com/github/TheAlgorithms/Python/builds/154205899#L1049

File: maths/allocation_number.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#-*- coding:utf-8 -*-
+
+
+def allocation_num(num, x) -> int:
```
Comment (cclauss) [inline maths/allocation_number.py]: Please see the __mypy__ output in the Travis CI log
https://github.com/TheAlgorithms/Python/blob/master/Travis_CI_tests_are_failing.md

**Comment by wind-Lv** [comment]
> https://travis-ci.com/github/TheAlgorithms/Python/builds/154205899#L1049

what's the problem,I try my best

**Comment by cclauss** [comment]
> what's the problem,I try my best

Thanks.  Keep at it.  You can do this.
https://travis-ci.com/github/TheAlgorithms/Python/builds/154205899#L1059-L1063 Says that the
when we run:
```
>>> f'{shares*3+1}-{shares*4}'
Expected:
    '12484-16647'
Got:
    '12484-16644'
```
So... What is `shares*4`?  Is it `16647` or `16644` ?  Does our test contain the right value?

You can run the doctests locally on your own computer with:
`python3 -m doctest -v maths/allocation_content_length.py`

**Comment by wind-Lv** [comment]
> > 有什么问题，我会尽力
> 
> 谢谢。继续吧 你可以这样做。
> https://travis-ci.com/github/TheAlgorithms/Python/builds/154205899#L1059-L1063说，
> 当我们运行时：
> 
> ```
> >>> f'{shares*3+1}-{shares*4}'
> Expected:
>     '12484-16647'
> Got:
>     '12484-16644'
> ```
> 
> 那么...是什么`shares*4`？是`16647`还是`16644`？我们的测试是否包含正确的值？
> 
> 您可以使用以下方法在自己的计算机上本地运行doctest：
> `python3 -m doctest -v maths/allocation_content_length.py`

The first problem was solved. No error is reported in doctest. In your test, share * 4 should not be executed because the for loop has exited.

**Comment by wind-Lv** [comment]
And I can't open travis-ci.com. If there is any mistake, please tell me directly, thank you.

File: maths/allocation_number.py (hunk 1 above)
Comment (cclauss) [reply maths/allocation_number.py]: Line 4 says that the function returns an `int` but line 26 says that we expect a `list` of `str`.  Line 26 is correct and line 4 is wrong.

$ __mypy --ignore-missing-imports .__
```
maths/allocation_number.py:51: error: Incompatible return value type (got "List[str]", expected "int")
Found 1 error in 1 file (checked 95 source files)
The command "mypy --ignore-missing-imports ." exited with 1.
```

**Pull request merged**